components/browser/state/src/main/java/mozilla/components/browser/state/ext/PermissionRequest.kt

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package mozilla.components.browser.state.ext
+import mozilla.components.concept.engine.permission.PermissionRequest
+
+/**
+ * An improved version of [List.contains] built to be used in [ContentStateReducer] to make sure
+ * no duplicate requests are added to the state.
+ * */
+fun List<PermissionRequest>.doesContain(permissionRequest: PermissionRequest): Boolean {
```

Comment:
I tried this initially but Android Studio tells me that "extension is shadowed by a member", so using my extension `.contains` will actually default to the standard `contains()`.

Reply:
I think we have to use the override operator keyword in front of the extension, maybe this example could [help](https://github.com/mozilla-mobile/android-components/blob/5cd371eebc9e2dfef8cc45472dd52a72b5a36558/components/concept/fetch/src/main/java/mozilla/components/concept/fetch/Headers.kt#L39)

Reply:
`operator fun List<PermissionRequest>.contains(permissionRequest: PermissionRequest): Boolean{
    return this.any {
        it.uri == permissionRequest.uri &&
                it.permissions == permissionRequest.permissions
    }
}`
This is what I've tried in my extension file, but I get the warning/error I mentioned. The example you've given is from a Iterable class, PermissionRequest isn't that type of class.

I can add the following snippet to the PermissionRequest class, 
`operator fun contains(list:List<PermissionRequest>):Boolean{
        return list.any {
            it.uri == this.uri &&
                    it.permissions == this.permissions
        }
    }`

...but I don't really like the syntax because it would look like `permissionRequest contains List<PermissionRequest>` -> `list in request`

To make it work and look properly I believe I'd have to create a custom list class for List<PermissionRequest>.... but that defeats the purpose of simplicity. 
Please let me know if I'm missing something as I don't have much experience with operator overloading.

Reply:
Got it, the issue because we have an extension function :( , what do you think about `containsPermission` or `hasPermission` ?